### ISSUE TITLE
UnixSocketAddress toString problem

### ISSUE THREAD

**Issue by gregw** [jnr/jnr-unixsocket#19]
Using the TestClient and TestServer on 
```
Linux Tile440 3.19.0-28-generic #30-Ubuntu SMP Mon Aug 31 15:52:51 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux
```

on the client, printing out address correctly results in:
```
connected to [family=PF_UNIX path=/tmp/fubar.sock]
```

but on the server, if you print out the address from                
```
                int n = channel.read(buf);
                UnixSocketAddress remote = channel.getRemoteSocketAddress();
                System.err.printf("%x Read in %d bytes from %s%n",hashCode(),n,remote);
```
you get output like:
```
29ee9faa Read in 39 bytes from [family=PF_UNIX path=6�J]
```

**Comment by gregw** [comment]
Note that getLocalSocketAddress() provides a correct address (which should be the same as the remote??)

**Comment by headius** [comment]
Nice...some encoding problem with the test on the server. Perhaps it's trying to treat UTF-8 bytes as though they were UTF-16 or vice versa?

**Comment by headius** [comment]
Nice...some encoding problem with the test on the server. Perhaps it's trying to treat UTF-8 bytes as though they were UTF-16 or vice versa?

**Comment by headius** [comment]
Actually, I think this was fixed by 75e0a70091c1d63416b9e13c3f8eba8210266efa.
